In [2]:
import pandas as pd
import numpy as np 
import warnings
warnings.filterwarnings('ignore')

We have imported the necesary libraries,now we will load the dataset that we have extracted from various sources.

In [11]:
df=pd.read_excel(r'C:\Nimish\Capstone-phase1-data.xls')
df

State/UT  Year  population(total)  Area (km2)  Literacy rate  \
0         A& N ISLANDS  2001             356650        8249          81.30   
1       ANDHRA PRADESH  2001           75728401      275045          60.47   
2    ARUNACHAL PRADESH  2001            1098328       83743          54.34   
3                ASSAM  2001           26638600       78438          63.25   
4                BIHAR  2001           82879911       94163          47.00   
..                 ...   ...                ...         ...            ...   
415         TAMIL NADU  2012           73141715      130058          80.88   
416            TRIPURA  2012            3718947       10486          80.89   
417      UTTAR PRADESH  2012          202934265      240928          67.98   
418        UTTARAKHAND  2012           10279517       53483          79.22   
419        WEST BENGAL  2012           92460379       88752          76.88   

     total literate population  Attempt to murder  Murder  Rape  \
0                     253135.0                  0      13     3   
1                   39934323.0               1555    2602   871   
2                     484785.0                 53      83    33   
3                   14015354.0                481    1356   817   
4                   31109577.0               3419    3643   888   
..                         ...                ...     ...   ...   
415                 53274562.0               2954    1949   737   
416                  2897340.0                 79     124   229   
417                118889202.0               4811    4966  1963   
418                  7100774.0                161     217   148   
419                 63274096.0               2854    2252  2046   

     Kidnapping and abduction  ...  Cruelty by husband or his relatives  \
0                           2  ...                                    9   
1                        1182  ...                                 5791   
2                          83  ...                                   11   
3                        1480  ...                                 1248   
4                        2159  ...                                 1558   
..                        ...  ...                                  ...   
415                      1945  ...                                 1965   
416                       139  ...                                  858   
417                      8878  ...                                 7661   
418                       297  ...                                  368   
419                      5117  ...                                19865   

     Importation of girls from other countries  Other IPC crimes  \
0                                            0               323   
1                                            7             34344   
2                                            0               618   
3                                            0              9315   
4                                           83             36667   
..                                         ...               ...   
415                                          0            124176   
416                                          0              1742   
417                                          0             68538   
418                                          0              2118   
419                                         12             64482   

     Total IPC crimes  Total Crimes  tot_crimes_sc  total crime against women  \
0                 658          1386              0                       34.0   
1              130089        278982           2933                    13669.0   
2                2342          5243              0                      180.0   
3               36877         81418              6                     4048.0   
4               88432        189612           1303                     4489.0   
..                ...           ...            ...                  

Data for the population,area of each state and literacy rate of each state have been collected from the different websites manually. The data was available for the year #2001 and #2011 for population, area and literacy rate of each state. So, necassary calculations and assumptions were made in order to achive the data for the year #2002 to #2010 & year #2012. **For example Andhra Pradesh's population icreased around 11% from 2001 to 2011 so we increased around 1% for every year to achieve the same.** Several websites were used in order to fetch the data for the same. Some of them are:-

- https://en.wikipedia.org/wiki/List_of_states_in_India_by_past_population
- https://en.wikipedia.org/wiki/List_of_states_and_union_territories_of_India_by_area
- https://en.wikipedia.org/wiki/List_of_Indian_states_and_union_territories_by_literacy_rate
- https://socialjustice.gov.in/common/76669
- http://cyberjournalist.org.in/census/cenindia.html
- https://www.indiastat.com/specimen-tables/geographical-data
- https://www.wbhealth.gov.in/other_files/2007/14_5.html
- https://en.wikipedia.org/wiki/List_of_states_and_union_territories_of_India_by_population